### PR TITLE
replace all instance of "gensub" with gsub for compatibility

### PR DESCRIPTION
**Pull request by YiwenShaoStephen**

### CODE REVIEW

File: egs/chime4/s5_1ch/local/run_dnn.sh [hunk 1]
```diff
@@ -147,7 +147,7 @@ srcdir=exp/tri4a_dnn_tr05_multi_${train}
 acwt=0.1
 
 # First we generate lattices and alignments:
-# gawk must be installed to perform awk -v FS="/" '{ print gensub(".gz","","",$NF)" gunzip -c "$0" |"; }' in
```
Comment (danpovey) [inline egs/chime4/s5_1ch/local/run_dnn.sh]: you can actually remove these comments about gawk, this should no longer be necessary.

File: egs/wsj/s5/steps/nnet/make_denlats.sh [hunk 2]
```diff
@@ -155,7 +155,7 @@ if [ $sub_split -eq 1 ]; then
   for n in `seq $nj`; do
     [ ! -d $dir/lat$n ] && mkdir $dir/lat$n;
     cat $sdata/$n/feats.scp | \
-    awk -v dir=$dir -v n=$n '{ utt=$1; utt_noslash=gensub("/","_","g",utt);
+    awk -v dir=$dir -v n=$n '{ utt=$1; utt_noslash=utt; gsub("/","_",utt_noslash);
```
Comment (danpovey) [inline egs/wsj/s5/steps/nnet/make_denlats.sh]: did you test these on a mac?

Comment (YiwenShaoStephen) [reply egs/wsj/s5/steps/nnet/make_denlats.sh]: I tested this syntax on my ubuntu 16.04.

File: egs/chime4/s5_1ch/local/run_dnn.sh (hunk 1 above)
Comment (YiwenShaoStephen) [reply egs/chime4/s5_1ch/local/run_dnn.sh]: Do you mean remove the entire comment here or just the words about gawk?

Comment (danpovey) [reply egs/chime4/s5_1ch/local/run_dnn.sh]: about gawk.

**Pull request merged**